dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckDataRelationsValidationHook.java

```diff
@@ -146,177 +166,354 @@ public void validateEvent( ValidationErrorReporter reporter, Event event )
                 }
             }
         }
+    }
 
-        if ( !programHasOrgUnit( program, organisationUnit, context.getProgramWithOrgUnitsMap() ) )
+    private void validateProgramHasOrgUnit( ValidationErrorReporter reporter, Event event,
+        TrackerImportValidationContext context, OrganisationUnit organisationUnit, Program program )
+    {
+        if ( programDoesNotHaveOrgUnit( program, organisationUnit, context.getProgramWithOrgUnitsMap() ) )
         {
             reporter.addError( event, E1029, organisationUnit, program );
         }
+    }
 
-        validateEventCategoryCombo( reporter, event, program );
+    private void validateEventCategoryCombo( ValidationErrorReporter reporter,
```

Comment:
will `validateEventCategoryOptionCombo` be much more readable than `validateEventCategoryCombo` here?

Events categoryCombo is just program's categoryCombo.